src/database/mod.rs

```diff
@@ -288,6 +288,23 @@ impl Database {
             .collect::<Vec<_>>())
     }
 
+    pub async fn get_last_commitment_by_status(&self, status: Status) -> Option<(usize, Hash)> {
+        let query = sqlx::query(
+            r#"
+                SELECT leaf_index, commitment
+                FROM identities
+                WHERE status = $1
+                ORDER BY leaf_index DESC
+                LIMIT 1   
```

Comment:
```suggestion
                SELECT MAX(leaf_index)
                FROM identities
                WHERE status = $1
```

?

Didn't test it yet as I can't get the correct rust version

Reply:
Nevermind, that only gets index